server/api-manager.js

```diff
@@ -968,9 +971,26 @@ class MindbodyQueries {
     request.addAuth(this.authToken);
     this.requestNum++;
     let allPagePromises = [];
+
+    let makeRequest = () => {
+      return request.makeRequest();
+    };
+    //backoff function, backs off a preset amount
+    const backoff = (retries, fn, delay = this.initialDelay) => {
+      return fn().catch((err) => {
+        if (retries > 1 && !this.atLimit()) {
+          pause(delay).then(() => {
```

Comment:
Where is this pause function defined for you? I added a pause function to fix this (for me) in the extra-simple-backoff-code-for-testing branch. When I run this code, throws an error like:
"ReferenceError: pause is not defined
    at file:///Users/dbrewer/dev/mindbody-report-generation/server/api-manager.js:983:11
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Promise.all (index 141)"

Reply:
Yes we need the pause function declaration. The call to the pause call existed in the initial attempt to fix the ENOTFOUND error, but it was never defined, so there was never a delay between attempts. I believe once reimplementing this, things should work smoothly.

Reply:
It is very interesting to me that this was working at all for me with these differences 👀.

Reply:
Hmm, I am unsure where the pause definition was coming from for me, I may have some extra npm modules :/ or something. In hindsight I think this should probably actually be done with setTimout directly though... something like setTimout(backoff, delay)